src/processMetric/ConvertGraphData.tsx

```diff
@@ -228,3 +230,92 @@ function getOneSeries(
   series.push(ser);
   return series;
 }
+
+export function withAppMetric(
+  width: number,
+  data: PanelData,
+  timeRange: TimeRange,
+  name: string,
+  level: string,
+) {
+  const applicationMetrics = [
+    'Service in count',
+    'Service out count',
+    'Service in responsetime sum',
+    'Service out responsetime sum',
+    'http in responsetime sum',
+    'http out responsetime sum',
+    'jvm memory heap',
+    'jvm memory non heap',
+  ];
+  let withApp = false;
+  for (let i = 0; i < applicationMetrics.length; i++) {
+    if ((getApplicationSeries(width, data, timeRange, name, level, applicationMetrics[i]) !== undefined) === true) {
+      withApp = true;
+    };
+  }
+  return withApp;
+}
+export function withInfMetric(
+  width: number,
+  data: PanelData,
+  timeRange: TimeRange,
+  name: string,
+  level: string
+) {
+  const infrastructureMetrics = [
+    'CPU Usage',
+    'Memory Usage',
+    'Memory Saturation',
+    'Network receive total',
+    'Network transmit total',
+    'Network receive saturation',
+    'Network transmit saturation',
+    'Network receive errors',
+    'Network transmit errors'
+  ];
+  let withInf = false;
+  for (let i = 0; i < infrastructureMetrics.length; i++) {
+    if ((getInfrastructureSeries(width, data, timeRange, name, level, infrastructureMetrics[i]) !== undefined) === true) {
+      withInf = true;
+    };
+  }
+  return withInf;
+}
+
+export function calcMoy(
+  data: PanelData,
+  name: string,
+  level: string,
+  metric: string
+) {
+
+  const metricName = convertMetricName(metric);
+  // convert metric to promql metric name
+
+  let dataIndex = 0;
+  for (let i = 0; i < data.series.length; i++) {
+    const temp = data.series[i].name?.split(' ');
+    if (temp !== undefined) {
+      if (temp[0] === name) {
+        if (temp[1] === level) {
+          if (temp[2] === metricName) {
+            dataIndex = i;
+          }
+        }
+      }
+    }
+  }
```

Comment:
this is just a matter of taste, but I wanted to show you a maybe more elegant solution for what you are trying to accomplish here:
```js
const dataIndex = data.series.findIndex(s => s.name ? s.name === `${name} ${level} ${metricName}` : false);
```